pkg/server/environment_create.go

```diff
@@ -41,6 +42,33 @@ func (s *Server) environmentCreate(c *gin.Context) {
 		return
 	}
 
+	res := req.Environment.Resources
+	resRequest := v1.ResourceList{}
+	if res.CPU != "" {
+		cpu, err := resource.ParseQuantity(res.CPU)
+		if err != nil {
+			c.JSON(500, err)
+			return
+		}
+		resRequest[v1.ResourceCPU] = cpu
+	}
+	if res.Memory != "" {
+		mem, err := resource.ParseQuantity(res.Memory)
+		if err != nil {
+			c.JSON(500, err)
+			return
+		}
+		resRequest[v1.ResourceMemory] = mem
+	}
+	if res.GPU != "" {
+		gpu, err := resource.ParseQuantity(res.GPU)
+		if err != nil {
+			c.JSON(500, err)
+			return
+		}
+		resRequest["nvidia/gpu"] = gpu
+	}
+
```

Comment:
Can we move them into a function? This `environmentCreate` is crazy long.

Reply:
done